pkg/operator/ceph/cluster/controller.go

```diff
@@ -267,29 +267,30 @@ func (c *ClusterController) configureExternalCephCluster(namespace, name string,
 	// then populate clusterInfo
 	cluster.Info = populateExternalClusterInfo(c.context, namespace)
 
-	// Validate versions (local and external)
-	_, _, err = c.detectAndValidateCephVersion(cluster, cluster.Spec.CephVersion.Image)
-	if err != nil {
-		return fmt.Errorf("failed to detect and validate ceph version. %+v", err)
-	}
-
-	err = populateConfigOverrideConfigMap(cluster.context, namespace, cluster.ownerRef)
-	if err != nil {
-		return fmt.Errorf("failed to populate config override config map. %+v", err)
+	// Write connection info (ceph config file and keyring) for ceph commands
+	if cluster.Spec.CephVersion.Image == "" {
+		err = mon.WriteConnectionConfig(c.context, cluster.Info)
+		if err != nil {
+			logger.Errorf("failed to write config. Attempting to continue. %+v", err)
+		}
 	}
 
-	// Write the rook-ceph-config configmap (used by various daemons to apply config overrides)
-	// If we don't do this, daemons will never start, waiting forever for this configmap to be present
-	err = config.GetStore(cluster.context, namespace, &cluster.ownerRef).CreateOrUpdate(cluster.Info)
-	if err != nil {
-		return err
-	}
+	// Validate versions (local and external)
+	// If no image is specified we don't perform any checks
+	if cluster.Spec.CephVersion.Image != "" {
+		_, _, err = c.detectAndValidateCephVersion(cluster, cluster.Spec.CephVersion.Image)
+		if err != nil {
+			return fmt.Errorf("failed to detect and validate ceph version. %+v", err)
+		}
 
-	// Apply CRD ConfigOverrideName to the external cluster
-	err = config.SetDefaultConfigs(cluster.context, namespace, cluster.Info)
-	if err != nil {
-		// Mons are up, so something else is wrong
-		return fmt.Errorf("failed to set Rook and/or user-defined Ceph config options on the external cluster monitors. %+v", err)
+		// Write the rook-ceph-config configmap (used by various daemons to apply config overrides)
+		// If we don't do this, daemons will never start, waiting forever for this configmap to be present
+		//
+		// Only do this when doing a bit of management...
+		err = config.GetStore(cluster.context, namespace, &cluster.ownerRef).CreateOrUpdate(cluster.Info)
```

Comment:
Where is this done for the local cluster configuration? Would it make sense to factor it into a common place?

Reply:
It's done by the mon here: https://github.com/rook/rook/blob/master/pkg/operator/ceph/cluster/mon/mon.go#L751
I don't think we can easily move it out.